src/transpiling/mod.rs

```diff
@@ -308,9 +308,12 @@ fn transpile(
 
 #[derive(Default, Clone)]
 struct DiagnosticCollector {
-  diagnostics_cell: Rc<RefCell<Vec<SwcDiagnostic>>>,
+  diagnostics_cell: Lrc<RefCell<Vec<SwcDiagnostic>>>,
 }
 
+unsafe impl Send for DiagnosticCollector {}
+unsafe impl Sync for DiagnosticCollector {}
```

Comment:
we can feature gate these, but without `concurrent`, [`Send` and `Sync` are noops](https://github.com/swc-project/swc/blob/9b76783281af7c6b92e8185b44ebefae41648d3f/crates/swc_common/src/sync.rs#L69-L87): 
```rust
#[cfg(not(feature = "concurrent"))]
mod single {
    pub use once_cell::unsync::{Lazy, OnceCell};
    /// Dummy trait because swc_common is in single thread mode.
    pub trait Send {}
    /// Dummy trait because swc_common is in single thread mode.
    pub trait Sync {}

    impl<T> Send for T where T: ?Sized {}
    impl<T> Sync for T where T: ?Sized {}

    pub use std::{
        cell::{
            Ref as ReadGuard, RefMut as WriteGuard, RefMut as MappedWriteGuard,
            RefMut as LockGuard, RefMut as MappedLockGuard,
        },
        rc::{Rc as Lrc, Weak},
    };
}
```

Reply:
If someone is using `concurrent` then this not being thread safe will cause thread concurrency issues. Probably this will need a thread safe implementation when compiling with concurrent.